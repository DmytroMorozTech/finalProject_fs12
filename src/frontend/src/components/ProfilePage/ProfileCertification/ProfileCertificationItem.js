import React from 'react'
import style from './styles'
import CreateIcon from '@material-ui/icons/Create'
import Typography from '@material-ui/core/Typography'
import toggleModalAction from '../../../redux/Modal/modalActions'
import {useDispatch} from 'react-redux'
import convertLocalDateToString from '../../../utils/convertLocalDateToString'
import { EDIT_CERTIFICATION } from '../../../redux/Modal/modalTypes'

function ProfileCertificationItem (props) {
  const {name, expirationDate, issueDate, hasExpiryDate, issuingOrganization, credentialId, credentialUrl} = props.certification

  const classes = style()
  const dispatch = useDispatch()
  const issuedMonthAndYear = convertLocalDateToString(issueDate)
  const expirationMonthAndYear = hasExpiryDate ? convertLocalDateToString(expirationDate) : ''

  return (
    <div className={classes.content}>
      <div className={classes.school}>
        <Typography variant="body1" className={classes.schoolName}>{name}</Typography>
        <Typography>{issuingOrganization}</Typography>
        <Typography>Issued: {issuedMonthAndYear}</Typography>
        <Typography> {hasExpiryDate ? `Expires: ${expirationMonthAndYear}` : 'No Expiration Date'} </Typography>
        <Typography>Credential ID: {credentialId}</Typography>
        <Typography>Credential URL: {credentialUrl}</Typography>
      </div>
      <div className={classes.btnCreate} onClick={() =>
        dispatch(toggleModalAction({modalType: EDIT_CERTIFICATION, certification: props.certification}))
      }>
        <CreateIcon className={classes.createIcon}/>
      </div>
    </div>
  )
}
export default ProfileCertificationItem