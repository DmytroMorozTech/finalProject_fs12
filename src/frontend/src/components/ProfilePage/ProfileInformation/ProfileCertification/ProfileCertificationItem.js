import React from 'react'
import style from '../styles'
import CreateIcon from '@material-ui/icons/Create'
import Typography from '@material-ui/core/Typography'
import toggleModalAction from '../../../../redux/Modal/modalActions'
import {useDispatch} from 'react-redux'
import convertLocalDateToString from '../../../../utils/convertLocalDateToString'
import { EDIT_CERTIFICATION } from '../../../../redux/Modal/modalTypes'
import clsx from 'clsx'

function ProfileCertificationItem (props) {
  const {name, expirationDate, issueDate, hasExpiryDate,
    issuingOrganization, credentialId, credentialUrl} = props.certification
  const {isEditable} = props

  const classes = style()
  const dispatch = useDispatch()
  const issuedMonthAndYear = convertLocalDateToString(issueDate)
  const expirationMonthAndYear = hasExpiryDate ? convertLocalDateToString(expirationDate) : ''

  return (
    <div className={classes.content}>
      <div>
        <Typography variant="body1" className={classes.title}>{name}</Typography>
        <Typography>{issuingOrganization}</Typography>
        <Typography>Issued: {issuedMonthAndYear}</Typography>
        <Typography> {hasExpiryDate ? `Expires: ${expirationMonthAndYear}` : 'No Expiration Date'} </Typography>
        {credentialId && <Typography>Credential ID: {credentialId}</Typography>}
        {credentialUrl && <Typography>Credential URL: {credentialUrl}</Typography>}
      </div>
      <div
        id='toggleModal'
        className={clsx(classes.btnCreate, !isEditable && classes.hidden)}
        onClick={() => dispatch(toggleModalAction(
          {modalType: EDIT_CERTIFICATION, certification: props.certification}
        ))}
      >
        <CreateIcon className={classes.createIcon}/>
      </div>
    </div>
  )
}

export default ProfileCertificationItem
