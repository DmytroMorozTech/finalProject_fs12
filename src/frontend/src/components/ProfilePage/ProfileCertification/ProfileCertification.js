import React from 'react'
import style from './styles'
import Typography from '@material-ui/core/Typography'
import ProfileCertificationItem from './ProfileCertificationItem'
import { useDispatch } from 'react-redux'
import toggleModalAction from '../../../redux/Modal/modalActions'
import { ADD_NEW_CERTIFICATION } from '../../../redux/Modal/modalTypes'
import AddIcon from '@material-ui/icons/Add'

function ProfileCertification (props) {
  const certifications = props.certifications
  const classes = style()
  const dispatch = useDispatch()

  return (
    <div className={classes.root}>
      <div className={classes.header}>
        <Typography variant="h3">
          Certification
        </Typography>
        <div onClick={() =>
          dispatch(toggleModalAction({modalType: ADD_NEW_CERTIFICATION}))}>
          <AddIcon className={classes.createIcon}/>
        </div>
      </div>
      <div>
        {certifications &&
        certifications
          .sort((cert1, cert2) => cert2.issueDate.localeCompare(cert1.issueDate))
          .map(
            certification => <ProfileCertificationItem key={certification.id} certification = {certification}/>)}
      </div>
    </div>
  )
}
export default ProfileCertification