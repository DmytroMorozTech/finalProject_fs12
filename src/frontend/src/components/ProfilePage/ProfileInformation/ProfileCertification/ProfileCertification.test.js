import { shallow } from 'enzyme'
import * as redux from 'react-redux'
import React from 'react'
import ProfileCertification from './ProfileCertification'

jest.mock('../styles', () => () => ({}))

const certifications = [
  {
    'id': 1,
    'name': 'Spring data jpa',
    'issuingOrganization': 'DAN-IT',
    'hasExpiryDate': true,
    'issueDate': '2016-01-01',
    'expirationDate': '2026-01-01',
    'credentialId': 'PS35092',
    'credentialUrl': 'https://dan-it.com.ua'
  },
  {
    'id': 2,
    'name': 'On-site practice',
    'issuingOrganization': 'Unilever',
    'hasExpiryDate': false,
    'issueDate': '2017-07-08',
    'expirationDate': null,
    'credentialId': 'RQ3627AS',
    'credentialUrl': 'https://unilever.ua'
  },
  {
    'id': 3,
    'name': 'Spring data jpa',
    'issuingOrganization': 'DAN-IT',
    'hasExpiryDate': true,
    'issueDate': '2016-07-08',
    'expirationDate': '2025-07-08',
    'credentialId': 'MK98123HAP',
    'credentialUrl': 'https://dan-it.com.ua'
  },
  {
    'id': 4,
    'name': 'Mongo DB course',
    'issuingOrganization': 'DAN-IT',
    'hasExpiryDate': true,
    'issueDate': '2016-03-06',
    'expirationDate': '2023-03-02',
    'credentialId': 'MK98123HAP',
    'credentialUrl': 'https://dan-it.com.ua'
  },
  {
    'id': 5,
    'name': 'Financial accounting for entrepreneurs',
    'issuingOrganization': 'UDEMY',
    'hasExpiryDate': true,
    'issueDate': '2016-07-08',
    'expirationDate': '2025-07-08',
    'credentialId': 'FG8892348',
    'credentialUrl': 'https://www.udemy.com/'
  },
  {
    'id': 6,
    'name': 'Soft skills upgrade',
    'issuingOrganization': 'UDEMY',
    'hasExpiryDate': true,
    'issueDate': '2016-03-06',
    'expirationDate': '2023-03-02',
    'credentialId': 'JK9928402',
    'credentialUrl': 'https://www.udemy.com/'
  },
  {
    'id': 7,
    'name': 'Interpersonal communications',
    'issuingOrganization': 'UDEMY',
    'hasExpiryDate': true,
    'issueDate': '2016-07-08',
    'expirationDate': '2025-07-08',
    'credentialId': 'OP2983849',
    'credentialUrl': 'https://www.udemy.com/'
  },
  {
    'id': 8,
    'name': 'Software development',
    'issuingOrganization': 'UDEMY',
    'hasExpiryDate': true,
    'issueDate': '2016-03-06',
    'expirationDate': '2023-03-02',
    'credentialId': 'MW99128430',
    'credentialUrl': 'https://www.udemy.com/'
  }
]

let isEditable

const setUp = (props) => shallow(<ProfileCertification {...props}/>)

describe('should render ProfileCertification component without props', () => {
  let spyOnUseDispatch
  let mockDispatch

  let emptyCertifications

  let component

  beforeEach(() => {
    spyOnUseDispatch = jest.spyOn(redux, 'useDispatch')
    mockDispatch = jest.fn()
    spyOnUseDispatch.mockReturnValue(mockDispatch)

    emptyCertifications = []

    isEditable = false

    component = setUp({emptyCertifications, isEditable})
  })
  it('should work dispatch', () => {
    expect(mockDispatch).toHaveBeenCalledTimes(0)
    component.find('div[id=\'toggleModal\']').simulate('click')
    expect(mockDispatch).toHaveBeenCalledTimes(1)
  })
})

describe('should render ProfileCertification component with props', () => {
  let spyOnUseDispatch
  let mockDispatch

  let component

  beforeEach(() => {
    spyOnUseDispatch = jest.spyOn(redux, 'useDispatch')
    mockDispatch = jest.fn()
    spyOnUseDispatch.mockReturnValue(mockDispatch)

    isEditable = true

    component = setUp({certifications, isEditable})
  })
  it('to match snapshot', () => {
    expect(component).toMatchSnapshot()
  })
})
