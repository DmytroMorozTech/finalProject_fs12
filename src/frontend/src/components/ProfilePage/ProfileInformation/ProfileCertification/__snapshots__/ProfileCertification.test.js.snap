// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ProfileCertification component with props to match snapshot 1`] = `
<div>
  <div>
    <WithStyles(ForwardRef(Typography))
      variant="h3"
    >
      Licenses & certifications
    </WithStyles(ForwardRef(Typography))>
    <div
      className=""
      id="toggleModal"
      onClick={[Function]}
    >
      <Memo(ForwardRef(AddIcon)) />
    </div>
  </div>
  <div>
    <ProfileCertificationItem
      certification={
        Object {
          "credentialId": "RQ3627AS",
          "credentialUrl": "https://unilever.ua",
          "expirationDate": null,
          "hasExpiryDate": false,
          "id": 2,
          "issueDate": "2017-07-08",
          "issuingOrganization": "Unilever",
          "name": "On-site practice",
        }
      }
      isEditable={true}
      key="2"
    />
    <ProfileCertificationItem
      certification={
        Object {
          "credentialId": "MK98123HAP",
          "credentialUrl": "https://dan-it.com.ua",
          "expirationDate": "2025-07-08",
          "hasExpiryDate": true,
          "id": 3,
          "issueDate": "2016-07-08",
          "issuingOrganization": "DAN-IT",
          "name": "Spring data jpa",
        }
      }
      isEditable={true}
      key="3"
    />
    <ProfileCertificationItem
      certification={
        Object {
          "credentialId": "FG8892348",
          "credentialUrl": "https://www.udemy.com/",
          "expirationDate": "2025-07-08",
          "hasExpiryDate": true,
          "id": 5,
          "issueDate": "2016-07-08",
          "issuingOrganization": "UDEMY",
          "name": "Financial accounting for entrepreneurs",
        }
      }
      isEditable={true}
      key="5"
    />
    <ProfileCertificationItem
      certification={
        Object {
          "credentialId": "OP2983849",
          "credentialUrl": "https://www.udemy.com/",
          "expirationDate": "2025-07-08",
          "hasExpiryDate": true,
          "id": 7,
          "issueDate": "2016-07-08",
          "issuingOrganization": "UDEMY",
          "name": "Interpersonal communications",
        }
      }
      isEditable={true}
      key="7"
    />
    <ProfileCertificationItem
      certification={
        Object {
          "credentialId": "MK98123HAP",
          "credentialUrl": "https://dan-it.com.ua",
          "expirationDate": "2023-03-02",
          "hasExpiryDate": true,
          "id": 4,
          "issueDate": "2016-03-06",
          "issuingOrganization": "DAN-IT",
          "name": "Mongo DB course",
        }
      }
      isEditable={true}
      key="4"
    />
    <ProfileCertificationItem
      certification={
        Object {
          "credentialId": "JK9928402",
          "credentialUrl": "https://www.udemy.com/",
          "expirationDate": "2023-03-02",
          "hasExpiryDate": true,
          "id": 6,
          "issueDate": "2016-03-06",
          "issuingOrganization": "UDEMY",
          "name": "Soft skills upgrade",
        }
      }
      isEditable={true}
      key="6"
    />
    <ProfileCertificationItem
      certification={
        Object {
          "credentialId": "MW99128430",
          "credentialUrl": "https://www.udemy.com/",
          "expirationDate": "2023-03-02",
          "hasExpiryDate": true,
          "id": 8,
          "issueDate": "2016-03-06",
          "issuingOrganization": "UDEMY",
          "name": "Software development",
        }
      }
      isEditable={true}
      key="8"
    />
    <ProfileCertificationItem
      certification={
        Object {
          "credentialId": "PS35092",
          "credentialUrl": "https://dan-it.com.ua",
          "expirationDate": "2026-01-01",
          "hasExpiryDate": true,
          "id": 1,
          "issueDate": "2016-01-01",
          "issuingOrganization": "DAN-IT",
          "name": "Spring data jpa",
        }
      }
      isEditable={true}
      key="1"
    />
  </div>
</div>
`;
